### DISCUSSION TITLE
Feature Request: Etsy Ads

### DISCUSSION THREAD

**Discussion by tverkade** [etsy/open-api#1082] [Ideas]
I would find it incredibly useful to be able to manage Etsy ads via the API. The ability to change the ad budget, view basic stats (such as Views, Clicks, Orders, Revenue, Spend), and toggle ads on/off for a specific listing would be great.

Personally, my main goal with this (other than just an easier & cleaner way to grab stats) would be implementing a per-listing ad budget, but simply disabling an ad once it reaches a set amount or dips below a certain ROAS value over a specific period of time.

**Comment by supermandee** [comment]
Is there an endpoint to show Etsy Ad spend currently? if not how can i infer it from other endpoints

**Reply by tverkade** [reply]
Nope. Nothing at all.

**Reply by tverkade** [reply]
@supermandee it doesn't even appear that a request to the v3/application/listings/{listing_id} endpoint will return whether or not that listing has ads enabled. Super disappointing.

**Reply by supermandee** [reply]
yeah im requesting access to etsy API but so far i haven't found anything related to etsy ads in the API endpoints, the only one i found for the "search_ads" in the billing endpoint was deprecated for v3

**Reply by kamaldon** [reply]
Ads endpoints be useful to be able to build tools like automatically updating the budget at certain times of the day, or for analytics.

**Reply by MichielDeRuiter** [reply]
Also, an endpoint for running sales would be amazing.

**Reply by Connor56** [reply]
Has anyone had a crack at doing this using the client-side API endpoints? I imagine it's possible to hack something together.

**Reply by turboturtleD** [reply]
> Has anyone had a crack at doing this using the client-side API endpoints? I imagine it's possible to hack something together.

Pretty sure what you are suggesting here violates the Etsy TOS.

**Reply by Connor56** [reply]
I haven't read them, but I imagine you're right. However, what other options are there if you want to review your data?

I notice there's no bulk download option for ad data, views, etc. Does the corporate API cover it?

**Reply by turboturtleD** [reply]
Likely no other options.  Just wanting anyone who reads the thread to understand the risk involved accessing the client API programmatically.

**Reply by Gareth064** [reply]
`review your data?`

I think what people often forget that with Etsy, all the data belongs to them. Sure they will allow you to bulk download your listings and orders. But the whole site is a walled garden for them to play in and we are just privileged guests who have no where else to go. This is why we will never get API endpoints to things like ads data, views and metrics. Because Etsy want to be the sole controller of it and doesn't want anyone outside of their system to benefit from it.

Once you come to realise that this API is not going to mature any further than it has, the happier life you will have :D

**Reply by etuannv** [reply]
> as anyone had a crack at doing this using the client-side API endpoints? I imagine it's possible to hack something toge

I’ve built a tool that automatically grabs Etsy ads data — kind of like a clever hack 😄.

**Reply by turboturtleD** [reply]
It’s considered screen scraping and violates the Etsy TOS.
